client/send-text.cc

```diff
@@ -42,19 +42,21 @@ static int usage(const char *progname) {
             "\t-o              : Only run once, don't scroll forever.\n"
             "\t-c<RRGGBB>      : Text color as hex (default: FFFFFF)\n"
             "\t-b<RRGGBB>      : Background color as hex (default: 000000)\n"
+            "\t-r              : Scroll text vertically (rotate)\n"
             );
 
     return 1;
 }
 
 int main(int argc, char *argv[]) {
     int width = 45;
-    int height = -1;
+    int height = 35;
```

Comment:
With this, you now switch off the automatic calculation of the height of the frame. It would've defaulted to the font-height before if this was found to be -1.

Now we essentially need to make both of these -1 initially. In the horizontal case, if we find width still to be -1, we set it to 45, the height to font.height().

In the vertical case, if width to be -1, we set it to the font-width (which might not be so easy to calculate, we could possibly find it with DrawText() of a "W" which is a wide character). If height is -1, then we set height to 35.